### ISSUE TITLE
`missing-argument` error for a Pydantic v1 `BaseModel` field that is `Optional`

### ISSUE THREAD

**Issue by juanzolotoochin** [astral-sh/ty#4349]
## Summary

`ty` 0.0.73 incorrectly reports a `missing-argument` error for a Pydantic v1 `BaseModel` field that is `Optional` but has no explicit default. Under Pydantic v1 semantics, an `Optional[...]` field with no explicit default implicitly defaults to `None`, so it should not be treated as required.

See https://docs.pydantic.dev/1.10/usage/models/#required-optional-fields

## Reproduction

`repro.py`:

```python
from typing import Optional

from pydantic.v1 import BaseModel


class Model(BaseModel):
    required: str
    implicit_default: Optional[str]
    explicit_default: Optional[str] = None


# Valid in Pydantic v1: Optional fields without an explicit default default to None.
Model(required="ok")
```

Run with `uv`

```bash
uvx --from ty==0.0.73 --with pydantic==2.13.4 ty check repro.py
```

## Actual output

```text
error[missing-argument]: No argument provided for required parameter `implicit_default`
  --> repro.py:13:1
   |
13 | Model(required="ok")
   | ^^^^^^^^^^^^^^^^^^^^

Found 1 diagnostic
```

## Expected output

No diagnostics. `Model(required="ok")` is valid under Pydantic v1: `implicit_default: Optional[str]` with no explicit default is treated by Pydantic v1 as optional, implicitly defaulting to `None` (unlike Pydantic v2, where an `Optional` field without a default is still required unless a default is given explicitly).

## Environment

- `ty` version: 0.0.73
- `pydantic` version: 2.13.4 (using the `pydantic.v1` compatibility shim)
- Python: 3.11+
- Installed via `uvx`, no project/Bazel/monorepo context required

## Notes

The `explicit_default` field (`Optional[str] = None`) is handled correctly and does not trigger a false positive — only the implicit-default case is affected. This suggests `ty`'s synthesized `__init__` signature for `pydantic.v1.BaseModel` is not applying v1's implicit-`None`-default rule for `Optional` fields.

### Version

 0.0.73

**Comment by carljm** [comment]
Hi, thanks for the report!

I don't think we plan to add support for behaviors specific to pydantic v1. Code using v1 can be checked, but we always use v2 semantics.
